barter-data-rs/src/dex/uniswapx/mod.rs

```diff
@@ -1,166 +1,192 @@
 
-      Ok(orders)
-  } else {
-      Err(DexError::Reqwest(response.error_for_status().unwrap_err()))
-  }
+        Ok(orders)
+    } else {
+        Err(DexError::Reqwest(response.error_for_status().unwrap_err()))
+    }
 }
 
 pub async fn get_order_by_hash(hash: String) -> Result<UniOrder, DexError> {
-  let url = format!("{}?orderHash={}", UNISWAPX_API, hash);
-  let response = reqwest::get(&url).await?;
+    let url = format!("{}?orderHash={}", UNISWAPX_API, hash);
+    let response = reqwest::get(&url).await?;
+
+    if response.status().is_success() {
+        let body: String = response.text().await?;
 
-  if response.status().is_success() {
-    let body: String = response.text().await?;
+        // Deserialize the JSON into the defined struct
+        let orders = deserialize_orders(&body).unwrap();
 
-    // Deserialize the JSON into the defined struct
-    let orders = deserialize_orders(&body).unwrap();
-    
-    if orders.len() == 1 {
-      Ok(orders[0].clone())
+        if orders.len() == 1 {
+            Ok(orders[0].clone())
+        } else {
+            Err(DexError::Error("UniOrder not found".to_owned()))
+        }
     } else {
-      Err(DexError::Error("UniOrder not found".to_owned()))
+        Err(DexError::Reqwest(response.error_for_status().unwrap_err()))
     }
-  } else {
-    Err(DexError::Reqwest(response.error_for_status().unwrap_err()))
-  }
 }
 
-
 // filter orders that don't already exist in self.open_orders
-pub fn filter_open_orders(open_orders: &Vec<UniOrder>, new_orders: &Vec<UniOrder>) -> Vec<UniOrder> {
-  let mut filtered_orders: Vec<UniOrder> = Vec::new();
-
-  for order in new_orders {
-    // us the order.order_hash to check if the order already exists in self.open_orders
-    let mut exists = false;
-    for open_order in open_orders {
-      if order.order_hash == open_order.order_hash {
-        exists = true;
-        break;
-      }
-    }
+pub fn filter_open_orders(
+    open_orders: &Vec<UniOrder>,
+    new_orders: &Vec<UniOrder>,
+) -> Vec<UniOrder> {
+    let mut filtered_orders: Vec<UniOrder> = Vec::new();
+
+    for order in new_orders {
+        // use the order.order_hash to check if the order already exists in self.open_orders
+        let mut exists = false;
+        for open_order in open_orders {
+            if order.order_hash == open_order.order_hash {
+                exists = true;
+                break;
+            }
+        }
 
-    if !exists {
-      filtered_orders.push(order.clone());
+        if !exists {
+            filtered_orders.push(order.clone());
+        }
     }
-  }
 
-  // return filtered orders
-  return filtered_orders
+    // return filtered orders
+    return filtered_orders;
 }
 
-#[derive(Debug, Clone)]
-pub struct UniswapX {}
-
-impl UniswapX {
-  pub fn new() -> Self {
-    UniswapX {
-    }
-  }
-
-  pub fn start(&self) -> UnboundedReceiver<Vec<IntentOrder>> {
+pub fn select() -> UnboundedReceiver<MarketEvent<DataKind>> {
```

Comment:
hm... select implies we're selecting a stream out of a set of streams, so the name might not be appropriate here.
maybe `init`?